src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -559,6 +560,24 @@ private void validate(CreateIndexClusterStateUpdateRequest request, ClusterState
             throw new IndexCreationException(new Index(request.index()),
                     new ElasticsearchIllegalArgumentException("custom data_paths for indices is disabled"));
         }
+
```

Comment:
I think I would move all settings validations including custom path settings validation above into a `validateIndexSettings` method and then we can call this method from the RestoreService. We will have an unnecessary validation for the number of shards there, but it's a small price to pay for not missing any other settings validations that will be added in the future.